src/client/Microsoft.Identity.Client.Broker/WamAdapters.cs

```diff
@@ -322,7 +322,9 @@ private static string GetExpectedRedirectUri(string clientId)
         {
             //MSAL Runtime throws an ApiContractViolation Exception with Tag: 0x2039c1cb (InvalidArg)
             //When no scopes are passed, this will check if user is passing scopes
-            if (!authenticationRequestParameters.HasScopes)
+            var scopes = authenticationRequestParameters.Scope.ElementAtOrDefault(0);
```

Comment:
`Scope` can be null, needs null check?

Reply:
if null scopes are passed then this will return null and the next check if catch it

Reply:
I mean if `Scope` itself is null, this will throw? `HasScopes` does a null check on `Scope`.

Reply:
```
            // Creating HashSet
            HashSet<string> myhash1 = new HashSet<string>();

            // Add the elements in HashSet
            //myhash1.Add("pass");
            myhash1.Add(null);
            //myhash1.Add(" ");
            //myhash1.Add("");

            var scopes = myhash1.ElementAtOrDefault(0);

            if (string.IsNullOrWhiteSpace(scopes))
            {
                Console.WriteLine("Scopes required!!!");
            }
            else
            {
                Console.WriteLine("Passed!!!");
            }
```

Reply:
I think we set the scopes to an empty sorted set if we detect that it is null so technically it should always be something. However, not sure if broker flow does something different. Doesn't hurt to add the check.